### PR TITLE
WIP: Basic UAVCAN integration

### PR DESCRIPTION
**Pull request by antoinealb**
This PR adds a basic UAVCAN node. For now it only publishes its node status on the bus using a fixed ID (42). I had to use some tricks in order to make libuavcan fit into this ressource constrained system:

* Implement some C++ stubs/syscalls to reduce flash footprint
* Basic C++ compiler tweaking such as disabling exceptions and RTTI
* Run the UAVCAN node in the main thread to avoid having a separate stack consuming RAM
* Disabling threadsafe statics to reduce flash footprint
* Using smaller buffers for UAVCAN queues (might be an issue, needs to be tested)
* Disabling UAVCAN assertions. This was the biggest gain (about 30kB of flash), but it suppresses some useful checks. In the future I would like to have a minimalistic handling of those assertions but for now it sounds good enough. Note that the assertions are still used when we don't use the bootloader so we can still do some basic sanity checking on hardware.

I still need to test that on hardware.

@nuft what do you think ?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -194,9 +220,15 @@ ULIBS =
 # End of user defines
 ##############################################################################
 
-RULESPATH = .
+RULESPATH = $(CHIBIOS)/os/common/ports/ARMCMX/compilers/gcc/
```
Comment (msplr) [inline Makefile]: typo here. should be `RULESPATH = $(CHIBIOS)/os/common/ports/ARMCMx/compilers/GCC`

**Comment by msplr** [comment]
good job 👍 

I'm worried that the binary size will augment later on, because now we don't use any UAVCAN messages - to be verified.

File: Makefile (hunk 1 above)
Comment (antoinealb) [reply Makefile]: Correct, I will fix it.

**Comment by antoinealb** [comment]
Ok, I changed the rules path (`GCC`) and rebased it in my commits. Can you merge it ?

Also for the binary size so far we are using the node status message (published periodically), so I think we should be fine. Should the size augment too much we will always find a way to make it fit.

**Pull request merged**